### PR TITLE
Actually add trailing commas to collection literals even if there are terminating comments

### LINKED ISSUE #3072: Trailing Comma in Dictionary not working if a comment line is at the end
**Describe the bug**

When running black on a dictionary with a comment, a trailing comma is not set correctly
The example 
```
{
    "a": fun(msg, "ts"),
    "longggggggggggggggid": ...,
    "longgggggggggggggggggggkey": ..., "created": ...
    # "longkey": ...
}
```
is being formatted to 
```
{
    "a": fun(msg, "ts"),
    "longggggggggggggggid": ...,
    "longgggggggggggggggggggkey": ...,
    "created": ...
    # "longkey": ...
}
```
where the `,` is missing after ` "created": ...`

**To Reproduce**

The bug can be reproduced using [black.vercel.app](https://black.vercel.app/?version=main&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ADmAHRdAD2IimZxl1N_Wm_vzTQ-P7e88-HPpthubCyLPup10IjvULL9Te6KZ1NodUgQMtz1NLUGAc5rjTKK_QogZJuLtXRH2sY5WOk6D72XAqkz1z1husCxzNbP5ioWHtGyI7H0HAVp5zntQmX8C62vRP-twtlIMnwtAB0bycSlKUlRAAGQAecBAACY-DmpscRn-wIAAAAABFla)

**Expected behavior**
The expected behavior would create a format with the trailing comma set at the end of the last dictionary entry:
```
{
    "a": fun(msg, "ts"),
    "longggggggggggggggid": ...,
    "longgggggggggggggggggggkey": ...,
    "created": ...,
    # "longkey": ...
}
```

**Environment**

- Black's version: 22.3.0 
- OS and Python version: Linux/Python 3.9.7 

**Additional context**

I had a look at the code yesterday, and I got the black dev environment locally up and running (great job with that btw, worked on the first try) and set up the test case that is failing, and  went into `black._format_str_once` with a debugger to understand what is happening.

I landed at `black.linegen.delimiter_split()` where I think the source of the issue lies, but I am not sure about that.

I got to this chunk of code, where the `current_line` is checked and always the last leave is selected and checked. If I understand correctly by some backwards looping through the line leaves.

image

When I let the debugger iterate and always check, what `current_line.leaves[-1]` prints, it always prints the last comment line `# "longkey": ...`. Not sure if I am on the right track though.

### PR DESCRIPTION
**Pull request by mainj12**
Fixes #3072 

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [ ] Add new / update outdated documentation?

### CODE REVIEW

File: tests/data/simple_cases/function_trailing_comma.py [hunk 1]
```diff
@@ -1,5 +1,11 @@
 def f(a,):
     d = {'key': 'value',}
+    e = {
```
Comment (JelleZijlstra) [inline tests/data/simple_cases/function_trailing_comma.py:3]: Can you add similar test cases using set, tuple, and list literals?

Comment (mainj12) [reply tests/data/simple_cases/function_trailing_comma.py:3]: Thanks for the comment. Yes, I've just added those.

File: tests/data/simple_cases/function_trailing_comma.py [hunk 2]
```diff
@@ -68,6 +92,34 @@ def f(
...
+        "longggggggggggggggid": ...,
+        "longgggggggggggggggggggkey": ...,
+        "created": ...,
+        # "longkey": ...
+    }
+    f = [
+        arg1,
+        arg2,
+        arg3,
+        arg4,
```
Comment (JelleZijlstra) [inline tests/data/simple_cases/function_trailing_comma.py:106]: This seems short enough to fit in one line, so it's a little weird to me that we don't. However, current behavior is already to put everything on its own line and that seems fine, so let's leave it as is.

**Review by JelleZijlstra — APPROVED** [review]

**Review by ichard26 — APPROVED** [review]
Thank you! This is great :ship:

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**